### PR TITLE
aws: different user agent on windows vs linux

### PR DESCRIPTION
**Pull request by PettitWesley**
Signed-off-by: Wesley Pettit <wppttt@amazon.com>

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Attached [local packaging test](./packaging/local-build-all.sh) output showing all targets (including any new ones) build.

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: src/aws/flb_aws_util.c [hunk 1]
```diff
@@ -47,6 +47,16 @@
 #define AWS_ECS_METADATA_URI "ECS_CONTAINER_METADATA_URI_V4"
 #define FLB_MAX_AWS_RESP_BUFFER_SIZE 0 /* 0 means unlimited capacity as per requirement */
 
+#ifdef FLB_SYSTEM_WINDOWS
+#define FLB_AWS_BASE_USER_AGENT        "aws-fluent-bit-plugin-windows"
+#define FLB_AWS_BASE_USER_AGENT_FORMAT "aws-fluent-bit-plugin-windows-%s"
+#define FLB_AWS_BASE_USER_AGENT_LEN    21
```
Comment (lubingfeng) [inline src/aws/flb_aws_util.c:53]: The len should be 29?

File: src/aws/flb_aws_util.c [hunk 2]
```diff
@@ -47,6 +47,16 @@
...
+#else
+#define FLB_AWS_BASE_USER_AGENT        "aws-fluent-bit-plugin"
+#define FLB_AWS_BASE_USER_AGENT_FORMAT "aws-fluent-bit-plugin-%s"
+#define FLB_AWS_BASE_USER_AGENT_LEN    29
```
Comment (lubingfeng) [inline src/aws/flb_aws_util.c:57]: the length should be 21 on Linux?

File: src/aws/flb_aws_util.c [hunk 3]
```diff
@@ -47,6 +47,16 @@
 #define AWS_ECS_METADATA_URI "ECS_CONTAINER_METADATA_URI_V4"
 #define FLB_MAX_AWS_RESP_BUFFER_SIZE 0 /* 0 means unlimited capacity as per requirement */
 
+#ifdef FLB_SYSTEM_WINDOWS
+#define FLB_AWS_BASE_USER_AGENT        "aws-fluent-bit-plugin-windows"
+#define FLB_AWS_BASE_USER_AGENT_FORMAT "aws-fluent-bit-plugin-windows-%s"
+#define FLB_AWS_BASE_USER_AGENT_LEN    21
+#else
+#define FLB_AWS_BASE_USER_AGENT        "aws-fluent-bit-plugin"
+#define FLB_AWS_BASE_USER_AGENT_FORMAT "aws-fluent-bit-plugin-%s"
```
Comment (lubingfeng) [inline src/aws/flb_aws_util.c:56]: How -%s is used in this?

Comment (PettitWesley) [reply src/aws/flb_aws_util.c:56]: See below, just fixed that

Comment (PettitWesley) [reply src/aws/flb_aws_util.c:56]: thanks

**Review by lubingfeng — APPROVED** [review]

File: src/aws/flb_aws_util.c (hunk 2 above)
Comment (PettitWesley) [reply src/aws/flb_aws_util.c]: yup fixed

File: src/aws/flb_aws_util.c (hunk 1 above)
Comment (PettitWesley) [reply src/aws/flb_aws_util.c]: yup fixed thanks

**Comment by PettitWesley** [comment]
@edsiper CI passes except for linter, can we merge?

**Pull request merged**